src/canvas.py

```diff
@@ -105,6 +105,23 @@ def __init__(self, application):
         self._legend_position = misc.LEGEND_POSITIONS[0]
         self._handles = []
 
+    def get_top_axis(self):
```

Comment:
I believe there is a logic problem with this function.

My guess is the intend is to loop over all items, figure out which axes are used and return the axes of the last item.

There are a few issues then:
using the whole application.get_data() ignores hide_uselected
this breaks when plotting more than one item
we are duplicating efforts already made in the item setter method.

If my assumption is true, I'd like to make a separate pr to integrate it into the item setter, as performance is critical here

Reply:
The intend was not to return the axis of the last item, but to return the top axis.

Basically, the axes have a hierarchy from bottom top top: axis, right_axis, top_left_axis to top_right_axis. The idea was to go through the axes in that order, and then if the axis is used, then it should be added to the list. It then returns the last axis in the list, which will always be the top one.

The problem is in the condition here. I wrote the loop here originally for the old style before we were using enums and then quickly adapted it, where I can see some issues. I'll go and edit it later this evening when I've got time. Basically the positions array should be four elements, (left, right, top, bottom), and then to get the list of used axis I can check if the corresponding directions are used to check if the axes should be considered to place the legend.

Reply:
This should work now